operator/cmd/mesh/manifest-apply.go

```diff
@@ -134,9 +140,19 @@ func InstallCmd(logOpts *log.Options) *cobra.Command {
 
 func runApplyCmd(cmd *cobra.Command, rootArgs *rootArgs, maArgs *manifestApplyArgs, logOpts *log.Options) error {
 	l := clog.NewConsoleLogger(cmd.OutOrStdout(), cmd.ErrOrStderr(), installerScope)
+	var opts clioptions.ControlPlaneOptions
+	kubeClient, err := kube.NewExtendedClient(kube.BuildClientCmd(maArgs.kubeConfigPath, maArgs.context), opts.Revision)
+	if err != nil {
+		return err
+	}
+	obv := version.OperatorBinaryVersion.String()
+	if err := DetectIstioVersionDiff(cmd, obv, kubeClient); err != nil {
```

Comment:
Looks like version striping does not return last part properly. I will verify this.

Reply:
I think it may be more fundamental problem that OperatorBinaryVersion != istio version mentioned below?

Reply:
`OperatorBinaryVersion` is useful in the following situation.

1. Users already have istio version `1.6.0` running in the cluster from the last 2 months where `OperatorBinaryVersion=1.6.0`
2. New version of istio `1.7.0` is released. The user wants to upgrade to the latest version where `OperatorBinaryVersion=1.7.0` 

In this scenario, different version detection will work properly as the comparison happens between `1.7.0 != 1.6.0` which is expected.

As `1.8.0` is not released, I think it only returns `1.8` from the control plan.

I have tested the comparison with `1.6.5` and it returns following.

```
$ $GOPATH/src/istio.io/istio/out/linux_amd64/istioctl install
! istioctl version 1.8.0 and the control plane version 1.6.5 are different.
This will install Istio 1.8.0 into the cluster. Proceed? (y/N)
```

Reply:
OperatorBinaryVersion is NOT the same as Istio version though, as far as I know. I don't agree with the entire operator versioning at all, but that is the way it is. For example, at `1.6.5` tag, `OperatorCodeBaseVersion = "1.6.0"`

Reply:
Got it, so with all the patch releases, `OperatorCodeBaseVersion` is not updated. I will revisit this and see if I can use tags.

Reply:
Don't check tags vs version either. We can only compare identical things. Tag != Version != OperatorCodeBaseVersion, none of the 3 should be compared against any others. Many users build custom images with things like 1.6.5-foobar. We can compare tags to tags though

Reply:
@howardjohn I have updated code to use the tag instead of version.

It uses [DockerBuildInfo.Tag](https://github.com/istio/pkg/blob/master/version/version.go#L59) for comparison and if the tag is `unknow` (in case of unreleased version 1.8) then it fallbacks to [OperatorVersionString](https://github.com/istio/istio/blob/master/operator/version/version.go#L30). Fallback is required otherwise it throws [this error](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/24263/integ-pilot-k8s-tests_istio/17166).